tests/TestRunner/upgrade_test.py

```diff
@@ -18,7 +18,7 @@
 HEALTH_CHECK_TIMEOUT_SEC = 5
 PROGRESS_CHECK_TIMEOUT_SEC = 30
 TESTER_STATS_INTERVAL_SEC = 5
-TRANSACTION_RETRY_LIMIT = 100
+TRANSACTION_RETRY_LIMIT = 1000 # TODO change back?
```

Comment:
I think something is up with the error handling/exponential backoff, because the test fails most times when this limit is 100.
Part of this may be that the blob granule client calls sometimes present errors or situations where it needs to retry to the client as transaction_too_old, just so that the retry loop will loop around again. There is either some issue with doing this or some issue with the exponential backoff loop.

Reply:
Ah I figured it out reading the onError code - we don't do backoff on transaction_too_old.
It sounds like it'd probably be best to add a blob granule - specific error and add it to retry loops. I don't think that would have any downgrade concerns, because the code running the onError code should be the same version as the client throwing it?

Reply:
Yes, I think if a workload needs 1000 retries to succeed, it is something that needs to be fixed. Probably it makes sense to introduce a new error code. The question is whether you want to make a separate PR for it...

Reply:
Yeah i'll make a separate PR

Reply:
https://github.com/apple/foundationdb/pull/8023

Reply:
Once this PR is merged, I'll rebase this PR on top of it, and change the limit back and ensure it passes the test with a limit of 100